packages/paste-core/components/blockquote/__tests__/index.spec.tsx

```diff
@@ -0,0 +1,102 @@
+      </Blockquote>,
+    );
+    const { getByTestId: idWithoutURL } = render(
+      <Blockquote data-testid="blockquoteWithoutURL">
+        <BlockquoteContent>This is some text.</BlockquoteContent>
+        <BlockquoteSource author="Google" source="People + AI Guidebook" />
+      </Blockquote>,
+    );
+
+    expect(getByTestId("blockquote").getAttribute("data-paste-element")).toEqual("BLOCKQUOTE");
+    expect(getByTestId("blockquote").querySelector(`[data-paste-element='BLOCKQUOTE_ICON']`)).toBeTruthy();
+    expect(getByTestId("blockquote").querySelector(`[data-paste-element='BLOCKQUOTE_CONTENT']`)).toBeTruthy();
+    expect(getByTestId("blockquote").querySelector(`[data-paste-element='BLOCKQUOTE_SOURCE']`)).toBeTruthy();
+    expect(getByTestId("blockquote").querySelector(`[data-paste-element='BLOCKQUOTE_SOURCE_AUTHOR']`)).toBeTruthy();
+    expect(getByTestId("blockquote").querySelector(`[data-paste-element='BLOCKQUOTE_SOURCE_CITE']`)).toBeTruthy();
+    expect(getByTestId("blockquote").querySelector(`[data-paste-element='BLOCKQUOTE_SOURCE_ANCHOR']`)).toBeTruthy();
+    expect(
+      idWithoutURL("blockquoteWithoutURL").querySelector(`[data-paste-element='BLOCKQUOTE_SOURCE_TEXT']`),
+    ).toBeTruthy();
+  });
+
+  it("should set custom element data attribute", () => {
+    const { getByTestId } = render(
+      <Blockquote url="#" data-testid="blockquote" element="CUSTOMIZED">
+        <BlockquoteContent element="CUSTOMIZED_CONTENT">This is some text.</BlockquoteContent>
+        <BlockquoteSource author="Google" source="People + AI Guidebook" element="CUSTOMIZED_SOURCE" />
+      </Blockquote>,
+    );
+    const { getByTestId: idWithoutURL } = render(
+      <Blockquote data-testid="blockquoteWithoutURL" element="CUSTOMIZED">
+        <BlockquoteContent element="CUSTOMIZED_CONTENT">This is some text.</BlockquoteContent>
+        <BlockquoteSource author="Google" source="People + AI Guidebook" element="CUSTOMIZED_SOURCE" />
+      </Blockquote>,
+    );
+
+    expect(getByTestId("blockquote").getAttribute("data-paste-element")).toEqual("CUSTOMIZED");
+    expect(getByTestId("blockquote").querySelector(`[data-paste-element='CUSTOMIZED_ICON']`)).toBeTruthy();
+    expect(getByTestId("blockquote").querySelector(`[data-paste-element='CUSTOMIZED_CONTENT']`)).toBeTruthy();
+    expect(getByTestId("blockquote").querySelector(`[data-paste-element='CUSTOMIZED_SOURCE']`)).toBeTruthy();
+    expect(getByTestId("blockquote").querySelector(`[data-paste-element='CUSTOMIZED_SOURCE_AUTHOR']`)).toBeTruthy();
+    expect(getByTestId("blockquote").querySelector(`[data-paste-element='CUSTOMIZED_SOURCE_CITE']`)).toBeTruthy();
+    expect(getByTestId("blockquote").querySelector(`[data-paste-element='CUSTOMIZED_SOURCE_ANCHOR']`)).toBeTruthy();
+    expect(
+      idWithoutURL("blockquoteWithoutURL").querySelector(`[data-paste-element='CUSTOMIZED_SOURCE_TEXT']`),
+    ).toBeTruthy();
+  });
```

Comment:
nit: these tests can be combined

Reply:
I simplified them a bit - want me to do more?